client/blocks/login/social-connect-prompt.jsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { capitalize } from 'lodash';
+import SocialLogo from 'social-logos';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import {
+	getLinkingSocialToken,
+	getLinkingSocialService,
+	getRedirectTo,
+} from 'state/login/selectors';
+import { connectSocialUser } from 'state/login/actions';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+class SocialConnectPrompt extends Component {
+	static propTypes = {
+		linkingSocialToken: PropTypes.string,
+		linkingSocialService: PropTypes.string,
+		onSuccess: PropTypes.func.isRequired,
+		redirectTo: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.handleClick = this.handleClick.bind( this );
+	}
```

Comment:
We can get rid of this constructor by using the arrow notation for the event handler:

```javascript
handleClick = ( event ) => {
  ...
};
```